nordicsemi/__main__.py

```diff
@@ -47,29 +59,18 @@
 import re
 sys.path.append(os.getcwd())
```

Comment:
This `sys.path` manipulation makes me think it's possible that moving the imports imports that were originally below will break some module loading. We need a comment on why we are adding the working directory to path. I hope somebody more familiar with the different uses / packaging of this tool can comment on this.

Reply:
From my experience, something like `sys.path.append(os.getcwd())` is only needed if you try to integrate other libs by adding their sources to your project, since `from nordicsemi` will otherwise not be found.

If you install the lib properly via `setup.py` (or from pypi) this is not needed. Therefore the code worked for me, even though my linter moved the sys.path at the end. 

Disclaimer: I am no python pro, therefore i might miss something here.

Reply:
So why might this be still useful, even though it is dangerous?
Suppose you clone the repo and trying to use the tool without installing it from another folder, then the exception `ModuleNotFound` will occur, since there is no `nordicsemi` folder in your current directory.

Reply:
It looks to me like it's just a result of a sloppy fix for the import error you get if you try to run `python nordicsemi/__main__.py`, as prompted by the readme in this repo. I won't blame whoever did this, since i think `sys.path` is a disaster in Python. `python nordicsemi/__main__.py`, `python nordicsemi` and `python -m nordicsemi` all run with different `sys.path`. At least `-m` is consistent with importing but then you again rely on the current working directory being where the module is located. Still I think depending on the working directory to be somewhere specific in relation to the executable or assets is bad design. I think we should change the readme to say to run `python nordicsemi`, which I think is the most reasonable.